export.py

```diff
@@ -0,0 +1,228 @@
+
+##########
+Example usage:
+python export.py --checkpoint ./checkpoints/yolact_darknet53_ks.pth \
+    --recipe ./recipes/yolact_ks.yaml \
+    --config yolact_darknet53_config
+
+##########
+Example Two:
+python export.py --checkpoint ./quantized-checkpoint/yolact_darknet53_1_10.pth \
+    --recipe ./recipes/yolact.quant.yaml \
+    --save-dir ./exported-models \
+    --name yolact_darknet53_quantized \
+    --batch-size 1 \
+    --image-shape 3 550 550 \
+    --config yolact_darknet53_config
+"""
+
+import argparse
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterable
+import torch
+from data import set_cfg
+from yolact import Yolact
+from sparseml.pytorch.utils import export_onnx
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+@dataclass
+class ExportArgs:
+    """
+    Typed arguments for exporting a yolact model to ONNX
+    """
+
+    checkpoint: Path
+    config: str
+    recipe: str
+    no_qat: bool
+    batch_size: int
+    image_shape: Iterable
+    save_dir: Path
+    name: Path
+
+    def __post_init__(self):
+        """
+        post-initialization processing and validation
+        """
+        self.checkpoint = Path(self.checkpoint)
+
+        if not self.checkpoint.exists():
+            raise FileNotFoundError(
+                f"The checkpoint {self.checkpoint} does " f"not exist."
+            )
+
+        self.image_shape = tuple(self.image_shape)
+        self.save_dir = Path(self.save_dir)
+        self.save_dir.mkdir(parents=True, exist_ok=True)
+        self.name = self.get_safe_name()
+
+    def get_safe_name(self):
+        if self.name:
+            self.name = Path(self.name)
+        else:
+            self.name = Path(self.checkpoint.with_suffix(".onnx").name)
+
+        filename = self.name.stem
+        self.name = self.save_dir / f"{filename}.onnx"
+        existence_counter = 0
+        while self.name.exists():
+            existence_counter += 1
+            self.name = self.name.parent / f"{filename}-{existence_counter}.onnx"
+        return self.name
+
+
+def parse_args() -> ExportArgs:
+    """
+    Add and parse arguments for exporting a yolact model to ONNX
+
+    :return: A ExportArgs Object with typed arguments
+    """
+    parser = argparse.ArgumentParser(description="Export yolact models to ONNX")
+
+    parser.add_argument(
+        "--checkpoint",
+        type=str,
+        required=True,
+        help="The yolact pytorch checkpoint to export",
+    )
+
+    parser.add_argument(
+        "--config",
+        "-c",
+        type=str,
+        default="yolact_base_config",
+        help="The config used to train the yolact model, for ex: "
+        "yolact_darknet53_config, yolact_resnet50_config, etc...; "
+        "Defaults to yolact_base_config.",
+    )
+
+    parser.add_argument(
+        "--recipe",
+        "-r",
+        type=str,
+        default=None,
+        help="Path or SparseZoo stub to the recipe used for training, "
```

Comment:
add that if no recipe is provided, but the checkpoint has one, that will be used. also that whichever recipe is used, will be applied to the model